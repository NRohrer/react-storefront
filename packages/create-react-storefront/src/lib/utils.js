/**
 * @license
 * Copyright Â© 2017-2019 Moov Corporation.  All rights reserved.
 */
const path = require('path')

/**
 * Return the target path to the React Storefront directory.
 *
 * @param {Object} userConfig The user's selected configuration options.
 */
const calculateReactStorefrontPath = (projectName, userConfig) => {
  let targetPath = process.cwd()
  if (userConfig.createDirectory) {
    targetPath = path.resolve(targetPath, projectName)
  }
  return targetPath
}

module.exports = {
  calculateReactStorefrontPath
}
