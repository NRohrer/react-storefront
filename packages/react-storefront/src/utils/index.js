/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */

// @create-index

export { default as batchPromises } from './batchPromises.js';
export { default as browser } from './browser.js';
export { default as createGenerateClassName } from './createGenerateClassName.js';
export { default as dataProps } from './dataProps.js';
export { default as hydrate } from './hydrate.js';
export { default as lazyLoadImages } from './lazyLoadImages.js';
export { default as minifyStyles } from './minifyStyles.js';
export { default as optimizeImages } from './optimizeImages.js';
export { default as scriptInjector } from './scriptInjector.js';
export { default as state } from './state.js';
export { default as url } from './url.js';

