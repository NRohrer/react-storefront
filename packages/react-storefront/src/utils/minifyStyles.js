/**
 * @license
 * Copyright Â© 2017-2019 Moov Corporation.  All rights reserved.
 */

import postcss from 'postcss'
import autoprefixer from 'autoprefixer'
import CleanCSS from 'clean-css'

const prefixer = postcss([autoprefixer])
const cleanCSS = new CleanCSS()

/**
 * Add prefixes and minify given CSS
 */
export default async function minifyStyles(css) {
  const prefixed = await prefixer.process(css, { from: undefined })
  return cleanCSS.minify(prefixed.css).styles
}
