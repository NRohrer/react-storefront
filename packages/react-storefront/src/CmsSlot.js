/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component, createRef } from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'
import { lazyLoadImages } from './utils/lazyLoadImages'

export const styles = theme => ({
  inline: {
    display: 'inline'
  },
  block: {
    display: 'block'
  },
  root: {
    '& .rsf-presized-img': {
      position: 'relative',

      '& img': {
        position: 'absolute',
        height: '100%',
        width: '100%',
        top: 0,
        left: 0
      }
    },
    '& img[data-rsf-lazy]': {
      visibility: 'hidden'
    }
  }
})

/**
 * A container for HTML blob content from a CMS.  Content is dangerously inserted into the DOM.
 * Pass the html as a string as the child of this component. Additional props are spread to the
 * rendered span element.
 */
@withStyles(styles, { name: 'RSFCmsSlot' })
export default class CmsSlot extends Component {
  static proptypes = {
    /**
     * Use inline prop to use display:inline style
     */
    inline: PropTypes.boolean,

    /**
     * Set to true to lazy load images that have been preprocessed with `$.lazyLoadImages()`.
     */
    lazyLoadImages: PropTypes.boolean
  }

  static defaultProps = {
    lazyLoadImages: false
  }

  constructor({ lazyLoadImages }) {
    super()

    if (lazyLoadImages) {
      this.el = createRef()
    }
  }

  componentDidMount() {
    if (this.props.lazyLoadImages) {
      lazyLoadImages(this.el.current)
    }
  }

  componentDidUpdate() {
    if (this.props.lazyLoadImages) {
      lazyLoadImages(this.el.current)
    }
  }

  render() {
    const { children, className, classes, inline, lazyLoadImages, ...others } = this.props

    return children ? (
      <span
        {...others}
        ref={this.el}
        className={classnames(className, classes.root, {
          [classes.inline]: inline,
          [classes.block]: !inline
        })}
        dangerouslySetInnerHTML={{ __html: children }}
      />
    ) : null
  }
}
