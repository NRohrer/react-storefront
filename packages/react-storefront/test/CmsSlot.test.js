/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import CmsSlot from '../src/CmsSlot'
import * as util from '../src/utils/lazyLoadImages'

describe('CmsSlot', () => {
  it('renders verbatim html', () => {
    const component = <CmsSlot className="slot1">{`<p>Content Here</p>`}</CmsSlot>

    expect(mount(component)).toMatchSnapshot()
  })

  it('renders nothing when no children are provided', () => {
    const component = <CmsSlot />

    expect(mount(component)).toMatchSnapshot()
  })

  it('passes onClick through to the underlying span', () => {
    const handler = jest.fn()

    mount(<CmsSlot onClick={handler}>{`Test`}</CmsSlot>).simulate('click')

    expect(handler).toHaveBeenCalled()
  })

  it('eager load lazy image source when InteractionObserver does not exist', () => {
    const wrapper = mount(
      <CmsSlot lazyLoadImages>
        {`
        <img data-src="foo" data-rsf-lazy>
        <img data-src="bar" data-rsf-lazy>
        `}
      </CmsSlot>
    )

    expect(wrapper.html()).toContain(' src="foo"')
    expect(wrapper.html()).toContain(' src="bar"')
  })

  it('lazy load images after prop has been updated', () => {
    const wrapper = mount(
      <CmsSlot lazyLoadImages>
        {`
        <img data-src="foo" data-rsf-lazy>
        <img data-src="bar" data-rsf-lazy>
        `}
      </CmsSlot>
    )

    const spy = jest.spyOn(util, 'lazyLoadImages')

    wrapper.setProps({
      children: `
      <img data-src="a" data-rsf-lazy>
      <img data-src="b" data-rsf-lazy>
      <img data-src="c" data-rsf-lazy>
    `
    })

    expect(spy).toBeCalledTimes(1)

    spy.mockRestore()
  })
})
