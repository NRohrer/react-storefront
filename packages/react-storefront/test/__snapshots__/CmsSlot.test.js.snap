// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CmsSlot renders nothing when no children are provided 1`] = `
<WithStyles(CmsSlot)>
  <CmsSlot classes={{...}} lazyLoadImages={false} />
</WithStyles(CmsSlot)>
`;

exports[`CmsSlot renders verbatim html 1`] = `
<WithStyles(CmsSlot) className="slot1">
  <CmsSlot className="slot1" classes={{...}} lazyLoadImages={false}>
    <span className="slot1 RSFCmsSlot-root-3 RSFCmsSlot-block-2" dangerouslySetInnerHTML={{...}} />
  </CmsSlot>
</WithStyles(CmsSlot)>
`;
